pcdsdevices/ipm.py

```diff
@@ -49,15 +96,276 @@ def removed(self):
+    delay = Cpt(EpicsSignal, ':TrigDelay', kind='config')
+    bias = Cpt(EpicsSignal, ':DiodeBias', kind='config')
+    ch0 = FCpt(IPIMBChannel, '{self.prefix}', channel_index=0, kind='normal')
+    ch1 = FCpt(IPIMBChannel, '{self.prefix}', channel_index=1, kind='normal')
+    ch2 = FCpt(IPIMBChannel, '{self.prefix}', channel_index=2, kind='normal')
+    ch3 = FCpt(IPIMBChannel, '{self.prefix}', channel_index=3, kind='normal')
+
+    # also be careful that you have to wait until trigger is enabled
+    # again before device can be used again.
+    # wait functionality?
+
+    def __init__(self, prefix, *, name, prefix_ioc, **kwargs):
+        if not prefix_ioc:
+            self._prefix_ioc = 'IOC:%s' % prefix
+        else:
+            self._prefix_ioc = prefix_ioc
+        super().__init__(prefix, name=name, **kwargs)
+
+    def screen(self):
+        """
+        Function to call the (pyQT) screen for an IPIMB box
+        """
+        executable = '/reg/g/pcds/controls/pycaqt/ipimb/ipimb.py'
+        if shutil.which(executable) is None:
+            print('%s is not on path, we cannot start the screen' % executable)
+            return
+        executable += ' --dir /reg/g/pcds/controls/pycaqt/ipimb'
+        os.system('%s --base %s --ioc %s --evr %s &' %
+                  (executable, self.prefix, self._prefix_ioc,
+                   self.prefix+':TRIG'))
+
+
+class Wave8Channel(Device):
+    """
+    Class for a single channel read out by a wave8
+
+    Parameters
+    ----------
+    prefix : ``str``
+        Wave8 base PV
+
+    name : ``str``
+        Alias for the wave8
+
+    channnel_index : ``int``
+        Index for gauge (0-15)
+    """
+
+    amplitude = FCpt(EpicsSignal, '{self.prefix}:AMPL_{self.channel_index}',
+                     kind='hinted')
+    tpos = FCpt(EpicsSignal, '{self.prefix}:TPOS_{self.channel_index}',
+                kind='normal')
+    number_of_samples = FCpt(
+        EpicsSignal, '{self.prefix}:NumberOfSamples{self.channel_index}_RBV',
+        write_pv='{self.prefix}:NumberOfSamples{self.channel_index}',
+        kind='config')
+    delay = FCpt(
+        EpicsSignal, '{self.prefix}:Delay{self.channel_index}', kind='config',
+        write_pv='{self.prefix}:Delay{self.channel_index}_RBV')
+
+    def __init__(self, prefix, *, name, channel_index,  **kwargs):
+        self.channel_index = channel_index
+        super().__init__(prefix, name=name, **kwargs)
+
+
+class Wave8(Device):
+    """
+    Class for a wave8
+
+    Parameters
+    ----------
+    prefix : ``str``
+        Wave8 base PV
+
+    name : ``str``
+        Alias for the wave8
+    """
+    isum = Cpt(EpicsSignal, ':SUM', kind='normal')
+    xpos = Cpt(EpicsSignal, ':XPOS', kind='normal')
+    ypos = Cpt(EpicsSignal, ':YPOS', kind='normal')
+    evr_channel = Cpt(Trigger, ':TRIG:TRIG0', kind='normal')
+    ch0 = FCpt(Wave8Channel, '{self.prefix}', channel_index=0, kind='normal')
```

Comment:
As no formatting is needed, these could all be `Cpt` with a suffix of `''`, i.e.,
```suggestion
    ch0 = Cpt(Wave8Channel, '', channel_index=0, kind='normal')
```